lib/internal/Magento/Framework/Encryption/Adapter/Mcrypt.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Magento\Framework\Encryption\Adapter;
+
+class Mcrypt implements EncryptionAdapterInterface
+{
+    /**
+     * @var string
+     */
+    private $cipher;
+
+    /**
+     * @var string
+     */
+    private $mode;
+
+    /**
+     * @var string
+     */
+    private $initVector;
+
+    /**
+     * @var resource
+     */
+    private $handle;
+
+    /**
+     * Mcrypt constructor.
+     * @param string $key
+     * @param string $cipher
+     * @param string $mode
+     * @param bool $initVector
+     * @throws \Exception
+     */
+    public function __construct(
+        $key,
+        $cipher = MCRYPT_BLOWFISH,
+        $mode = MCRYPT_MODE_ECB,
+        $initVector = false
+    ) {
+        $this->cipher = $cipher;
+        $this->mode = $mode;
+        // @codingStandardsIgnoreStart
+        $this->handle = @mcrypt_module_open($cipher, '', $mode, '');
+        // @codingStandardsIgnoreEnd
+        try {
+            // @codingStandardsIgnoreStart
+            $maxKeySize = @mcrypt_enc_get_key_size($this->handle);
+            // @codingStandardsIgnoreEnd
+            if (strlen($key) > $maxKeySize) {
+                throw new \Magento\Framework\Exception\LocalizedException(
+                    new \Magento\Framework\Phrase('Key must not exceed %1 bytes.', [$maxKeySize])
+                );
+            }
+            // @codingStandardsIgnoreStart
+            $initVectorSize = @mcrypt_enc_get_iv_size($this->handle);
+            // @codingStandardsIgnoreEnd
+            if (true === $initVector) {
+                /* Generate a random vector from human-readable characters */
+                $abc = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+                $initVector = '';
```

Comment:
Changing variable type from `bool` to `string` on the fly can be confusing and potentially lead to bugs.